src/main/java/org/codehaus/mojo/flatten/ElementHandling.java

```diff
@@ -47,6 +47,9 @@
     keep,
 
     /** Remove the element entirely so it will not be present in flattened POM. */
-    remove
+    remove,
+
+    /** Take the element untouched from the original POM. Fix for {@link #keep} */
+    keepRaw

```

Comment:
interesting. So actually `keep` does not actually do what it should? Can you give an example?
And you invented a new handling to avoid breaking existing build configs?
That would make a lot of sense. So does `keep` somehow already resolve variables or what is it that it does too much?

Reply:
The problem that arose for me was that the configuration in (active) profiles was interpolated, specifically `${basedir}`.

And I did not invent anything, just rebased it. ^^

Reply:
Without wanting to be rude, but the overall direction of the original PR-code is rather being a work-around that being a real fix :D

I guess the work remaining now is to fix the underlying issues so the workarounds (`version`, `keepRaw`) are not necessary anymore.